subprojects/dependency-management/src/testFixtures/groovy/org/gradle/resolve/scenarios/VersionRangeResolveTestScenarios.groovy

```diff
@@ -90,31 +110,32 @@ class VersionRangeResolveTestScenarios {
         conflicts: true
     ).and(
         versions: [PREFER_11, PREFER_10_12],
-        expected: "11",
+        expected: "12",
```

Comment:
What's the logic behind that behavior change? Why would we not be merging prefers when they are the only source?

Reply:
Two reasons:
1) It would be surprising if we merged `prefer` constraints in the absence of any `require` constraints, but then stopped merging as soon as a requirement on `(0,)` was added.
2) So far I haven't been able to come up with a reasonable algorithm for merging `prefer` constraints always (even in the presence of `require` constraints).

So I've removed the surprising/inconsistent behaviour, and later we can add the unsurprising behaviour (2) if we want.